website/docs/cdktf/concepts/best-practices.mdx

```diff
@@ -0,0 +1,30 @@
+---
+
+# Best Practices
+
+## Use `TerraformVariable` to read secrets
+
+If you read a secret from the environment or out of a file with normal file system access and you use them in your CDKTF code, they will end up in the synthesized Terraform code. The problem with that is that it might leak if an attacker can get access to your Terraform code. To mitigate this, you should [use the `TerraformVariable` construct](/cdktf/concepts/variables-and-outputs#input-variables) to read secrets.
+
+## Use pre-built provider bindings if possible
+
+Pre-built provider bindings are available for many popular providers, they have the same code as if you'd run `cdktf get` for this provider. [You can find a complete list through Github](https://github.com/topics/pre-built-provider). They help you by reducing the time `cdktf get` takes and they can be used as a peer dependency if you use open source constructs.
+
+## Build extensible constructs
+
+No matter if you are building constructs for your organization or as open source code: Constructs are a great way to abstract common behaviour into reusable classes. If you have no reason to limit the extensibility of a construct you should default to making it as easy to the user as possible to overwrite custom behaviour, while still providing good defaults for the standard behaviour.
+You can use interfaces from the generated provider bindings to allow users to overwrite configuration you write if the number of configurations is limited enough.
+In very complex constructs it can be useful to encapsulate overwritable behaviour into methods. This way a user can extend the base class and overwrite the behaviour in a single place.
+
+## Use projen to distribute constructs
+
+If your code is hosted on Github and you want to distribute it as a CDKTF construct, you can use [projen](https://github.com/projen/projen) to create a repository with all required tooling set up for you. You can run `npx projen new cdktf-construct` in a new folder and the created project will be ready to use. Projen has builtin options to [publish constructs to all registries](https://github.com/projen/projen/blob/main/src/cdk/jsii-project.ts#L48).
+
+If you want to deploy your CDKTF construct as a Terraform module we recommend checking out [`projen-cdktf-hybrid-construct`](https://github.com/DanielMSchmidt/projen-cdktf-hybrid-construct).
+
+## Seperate business units in stacks
+
+When it comes to the question where to split stacks we recommend creating one class that encapsulates the logic for all environments. An environment might be a deployment stage (development / staging / production), a region (eu-west-1 / us-east-1), or an application (e-commerce / blog / data warehouse). This way you share the benefit of quickly changing the logic between all stacks (you only need to write the code once), but you maintain a clear seperation and a small blast radius.
```

Comment:
> we recommend creating one class that encapsulates the logic for all environments.

I'm not sure if I really get what you mean with this. Having a single place (like `app.ts`) where all Stacks are instantiated?

The sentence I quoted could be misinterpreted as "put all logic [aka resources] into a single class".